ruby-challenges.rb

```diff
@@ -10,23 +10,85 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def letter_checker (array, string)
+    array.select do |value| 
+        value.include? string
+    end
+end
+
+p letter_checker(beverages_array, letter_o) #["coffee", "soda water"]
+p letter_checker(beverages_array, letter_t) #["tea", "water", "soda water"]
+
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+# pseudo:
+# what's needed: A method to take a hash and unpack it. Takes an array as argument
+# research 2d array, can I use flatten? ref: https://apidock.com/ruby/Array/flatten
+    # flatten works but gives me the keys as well as the values... flatten can take an argument. flatten(value?)
+    # Am I trying to flatten the hashes not the array... or both
+    # try array.values or hash.values ref: https://docs.ruby-lang.org/en/master/Hash.html#class-Hash-label-Methods+for+Converting
+# after unpacking sort alphabetically with .sort()
+# input: array of hashes
+# output: one dimentional array
+
+def crush_the_hash (array)
+    array.values.flatten.sort    
+end
```

Comment:
Sweet! The argument is not a array of hashes, but a hash with the values of the keys being arrays!

Reply:
thank you Chris. This had me confused, I'm not sure why I was seeing it as an array of hashes. Makes more sense now and I can better understand why I needed to do hash.values. It worked because the data type passed as an argument was a hash, so the method was fine despite me calling it an array. When I was doing research I was getting stuck because of that.